dkg/pedersen/handler.go

```diff
@@ -530,3 +607,299 @@ func (h *Handler) announceDkgPublicKey(out mino.Sender, from mino.Address) error
+			dela.Logger.Warn().Msgf(
+				"%v ignored reshare request from %v as it has"+
+					" started before\n", h.me, from)
+			return xerrors.Errorf("dkg is started")
+		}
+
+	} else { // if the node is in the old committee or is a common node
+		if !h.startRes.Done() {
+			dela.Logger.Warn().Msgf(
+				"%v ignored reshare request from %v as it is not"+
+					" started yet\n", h.me, from)
+			return xerrors.Errorf("dkg is not started")
+		}
+	}
+
+	if h.startRes.IsReshare() {
+		dela.Logger.Warn().Msgf(
+			"%v ignored reshare request from %v as it is already"+
+				" resharing\n", h.me, from)
+		return xerrors.Errorf("dkg is already resharing")
+	}
+
+	h.startRes.Reshare()
+
+	ctx := context.Background()
+	ctx, _ = context.WithTimeout(ctx, 5*time.Minute)
+	go func() {
+		err := h.reshare(ctx, msg, from, out)
+		if err != nil {
+			dela.Logger.Error().Msgf(
+				"%v failed to reshare: %v", h.me, err)
+			h.startRes.AbortReshare()
+		}
+		close(done)
+	}()
+
+	return nil
+}
+
+// reshare is called when the node has received its reshare message. Note that we
+// might have already received some deals from other nodes in the meantime. The
+// function handles the DKG resharing protocol.
+func (h *Handler) reshare(ctx context.Context, resharingRequest types.ResharingRequest,
+	from mino.Address, out mino.Sender) error {
+
+	dela.Logger.Info().Msgf("%v is resharing", h.me)
+
+	addrsNew := resharingRequest.GetAddrsNew()
+
+	if len(addrsNew) != len(resharingRequest.GetPubkeysNew()) {
+		return xerrors.Errorf(
+			"there should be as many participants as "+
+				"pubKey: %d != %d", len(addrsNew),
+			len(resharingRequest.GetPubkeysNew()),
+		)
+	}
+	// by default the node is not common. later we check
+	commonNode := false
+
+	// if the node is in the old committee, it should do the following steps
+	if h.startRes.distrKey != nil {
+		addrsOld := h.startRes.GetParticipants()
+
+		// this variable is true if the node is common between the old and the new committee
+		commonNode = addrIsInSlice(h.me, addrsNew) && addrIsInSlice(h.me, addrsOld)
+
+		// 1. update mydkg for resharing
+		share, err := h.dkg.DistKeyShare()
+		if err != nil {
+			return xerrors.Errorf("failed to create : %v", err)
+		}
+		c := &pedersen.Config{
+			Suite:        suite,
+			Longterm:     h.privKey,
+			OldNodes:     h.startRes.GetPublicKeys(),
+			NewNodes:     resharingRequest.GetPubkeysNew(),
+			Share:        share,
+			Threshold:    resharingRequest.GetTNew(),
+			OldThreshold: h.startRes.threshold,
+		}
+		d, err := pedersen.NewDistKeyHandler(c)
+		if err != nil {
+			return xerrors.Errorf("failed to compute the new dkg: %v", err)
+		}
+		h.dkg = d
+
+		// 2. Send my Deals to the new and common nodes
+		err = h.sendDealsResharing(ctx, out, addrsNew, share.Commits)
+		if err != nil {
+			return xerrors.Errorf("failed to send deals: %v", err)
+		}
+	}
+	// if the node is a new node or is a common node it should do the next steps
+	if h.startRes.distrKey == nil || commonNode {
```

Comment:
I'm wondering if it wouldn't be easier to define variables for these conditions, i.e. `isOldNode := h.startRes.distrKey != nil` , `isCommonNode`,  `isNewNode := !isOldNode`
it might make the conditions more readable throughout this function, and remove the need for comments to clarify the conditions themselves.